frame/assets/src/lib.rs

```diff
@@ -247,19 +256,594 @@ decl_error! {
+		/// over this asset, but may later change and configure the permissions using `transfer_ownership`
+		/// and `set_team`.
+		/// - `max_zombies`: The total number of accounts which may hold assets in this class yet
+		/// have no existential deposit.
+		/// - `min_balance`: The minimum balance of this new asset that any single account must
+		/// have. If an account's balance is reduced below this, then it collapses to zero.
+		///
+		/// Emits `Created` event when successful.
+		///
+		/// Weight: `O(1)`
+		#[weight = T::WeightInfo::create()]
+		fn create(origin,
+			#[compact] id: T::AssetId,
+			admin: <T::Lookup as StaticLookup>::Source,
+			max_zombies: u32,
+			min_balance: T::Balance,
+		) {
+			let owner = ensure_signed(origin)?;
+			let admin = T::Lookup::lookup(admin)?;
+
+			ensure!(!Asset::<T>::contains_key(id), Error::<T>::InUse);
+
+			let deposit = T::AssetDepositPerZombie::get()
+				.saturating_mul(max_zombies.into())
+			 	.saturating_add(T::AssetDepositBase::get());
+			T::Currency::reserve(&owner, deposit)?;
+
+			Asset::<T>::insert(id, AssetDetails {
+				owner: owner.clone(),
+				issuer: admin.clone(),
+				admin: admin.clone(),
+				freezer: admin.clone(),
+				supply: Zero::zero(),
+				deposit,
+				max_zombies,
+				min_balance,
+				zombies: Zero::zero(),
+				accounts: Zero::zero(),
+			});
+			Self::deposit_event(RawEvent::Created(id, owner, admin));
+		}
+
+		/// Issue a new class of fungible assets from a privileged origin.
+		///
+		/// This new asset class has no assets initially.
+		///
+		/// The origin must conform to `ForceOrigin`.
+		///
+		/// Unlike `create`, no funds are reserved.
+		///
+		/// - `id`: The identifier of the new asset. This must not be currently in use to identify
+		/// an existing asset.
+		/// - `owner`: The owner of this class of assets. The owner has full superuser permissions
+		/// over this asset, but may later change and configure the permissions using `transfer_ownership`
+		/// and `set_team`.
+		/// - `max_zombies`: The total number of accounts which may hold assets in this class yet
+		/// have no existential deposit.
+		/// - `min_balance`: The minimum balance of this new asset that any single account must
+		/// have. If an account's balance is reduced below this, then it collapses to zero.
+		///
+		/// Emits `ForceCreated` event when successful.
+		///
+		/// Weight: `O(1)`
+		#[weight = T::WeightInfo::force_create()]
+		fn force_create(origin,
+			#[compact] id: T::AssetId,
+			owner: <T::Lookup as StaticLookup>::Source,
+			max_zombies: u32,
+			min_balance: T::Balance,
+		) {
+			T::ForceOrigin::ensure_origin(origin)?;
+			let owner = T::Lookup::lookup(owner)?;
+
+			ensure!(!Asset::<T>::contains_key(id), Error::<T>::InUse);
+
+			Asset::<T>::insert(id, AssetDetails {
+				owner: owner.clone(),
+				issuer: owner.clone(),
+				admin: owner.clone(),
+				freezer: owner.clone(),
+				supply: Zero::zero(),
+				deposit: Zero::zero(),
+				max_zombies,
+				min_balance,
```

Comment:
```suggestion
				min_balance.max(1),
```

Reply:
might be worthwhile to make a `fn new` for this `AssetDetails` then.